portable/ARMv8M/non_secure/port.c

```diff
@@ -497,17 +496,29 @@ PRIVILEGED_DATA static volatile uint32_t ulCriticalNesting = 0xaaaaaaaaUL;
             __asm volatile ( "dsb" );
             __asm volatile ( "isb" );
 
-            /* Disable the SysTick clock without reading the
-             * portNVIC_SYSTICK_CTRL_REG register to ensure the
-             * portNVIC_SYSTICK_COUNT_FLAG_BIT is not cleared if it is set.  Again,
-             * the time the SysTick is stopped for is accounted for as best it can
+            /*
+             * Read the contents of the portNVIC_SYSTICK_CTRL_REG register into
+             * ulCtrlRegValue to ensure portNVIC_SYSTICK_COUNT_FLAG_BIT is taken
+             * into account later on.
+             */
+            ulCtrlRegValue = portNVIC_SYSTICK_CTRL_REG;
+
+            /* Disable the SysTick timer. */
+            ulCtrlRegValue &= ~( portNVIC_SYSTICK_ENABLE_BIT );
+            portNVIC_SYSTICK_CTRL_REG = ulCtrlRegValue;
+
+            /* Update ulCtrlRegValue */
+            ulCtrlRegValue |= portNVIC_SYSTICK_CTRL_REG;
```

Comment:
The value of the register is re-read to prevent a race condition while portNVIC_SYSTICK_ENABLE_BIT is still set.